### PR TITLE
DSD-1139: Secondary Button styles

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-1139](https://jira.nypl.org/browse/DSD-1139)

## This PR does the following:

- Updates the colors for the `secondary` type of the `Button` component.

## How has this been tested?

- local Storybook

## Accessibility concerns or updates

- The colors changes are using our WCAG compliant color palette.

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 
 - Adds the `text` variant in the `Button` component.
 
+### Updates
+
+- Updates the colors for the `secondary` type of the `Button` component.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: And `iconOnly` type?

File: src/theme/components/button.ts [hunk 2]
```diff
@@ -100,11 +103,12 @@ const pill = {
 const iconOnly = {
   bg: "ui.white",
   border: "1px solid",
-  borderColor: "ui.gray.light-cool",
-  color: "inherit",
+  borderColor: "ui.link.primary",
+  color: "ui.link.primary",
```
Comment (oliviawongnyc) [inline src/theme/components/button.ts:107]: Does inherit no longer work or is this an update for clarity?

Comment (bigfishdesign13) [reply src/theme/components/button.ts:107]: The color for this variant is changing altogether, so `inherit` will not work in this case.

File: CHANGELOG.md (hunk 1 above)
Comment (bigfishdesign13) [reply CHANGELOG.md]: This is interesting.  We have `iconOnly` as a unique variant on the same level as `primary`, `secondary`, etc.  However, `iconOnly` should be a state of a button type.  For example, we should be able to have an icon only `callout` button.  But right now, `iconOnly` is a variant that matches the `secondary` styles and that is the only type of icon onlu button.  That is why I wrote this as I did.

For now, I will update the CHANGELOG entry and I will also create a new ticket to refactor how `iconOnly` is handled.

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**